h/static/scripts/annotator/guest.coffee

```diff
@@ -6,10 +6,22 @@ module.exports = class Guest extends Annotator
 
   # Events to be bound on Annotator#element.
   events:
+    "setVisibleHighlights": "setVisibleHighlights"
     ".annotator-adder button click":     "onAdderClick"
     ".annotator-adder button mousedown": "onAdderMousedown"
     ".annotator-adder button mouseup":   "onAdderMouseup"
-    "setVisibleHighlights": "setVisibleHighlights"
+    "document onselectionchange": "userSelectionChanged"
+
+  touch = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)
+
+  # This hack is needed for iOS Safari only.
+  if /iPhone|iPad|iPod/i.test(navigator.userAgent)
+    $(window).bind 'selectionEnd', ->
+      # reset selection timeout
+      selectionEndTimeout = null
```

Comment:
These need to be defined somewhere and initialized to `null`, such as at the top of this module. One of the nasty side-effects of coffee is that `var` is no longer explicit.